pythia/seo/sunspotter.py

```diff
@@ -858,3 +864,94 @@ def get_lat_lon_in_hgs(self, obsdate, get_nearest=True):
+
+    def match_with_swpc_for_obsdate(self, obsdate, *, days_delta=1, match_type='euclidean',
+                                    noaa_ar='NOAA SWPC Observer', fmt='%Y-%m-%d %H:%M:%S', match_threshold=20):
+        """
+        Match Sunspotter observations to observations from HEK.
+
+        Parameters
+        ----------
+        obsdate : str
+            The observation time and date.
+        days_delta : int, optional
+            Number of days from the obsdate, by default 1
+        match_type : str, optional
+            The matching Algorithm that tablematcher will use, by default 'euclidean'
+        noaa_ar : str, optional
+            Parameter used by HEK attr, by default 'NOAA SWPC Observer'
+        fmt : str, optional
+            Format of the obsdate, by default '%Y-%m-%d %H:%M:%S'
+        match_threshold : float, optional
+            Threshold for being considered a good match
+
+        Returns
+        -------
+        compare_df : pd.DataFrame
+            Dataframe comparing the matched values
+
+        Examples
+        --------
+        >>> from pythia.seo import Sunspotter
+        >>> sunspotter = Sunspotter()
+        >>> obsdate = '2005-12-31 12:48:02'
+        >>> sunspotter.match_with_swpc_for_obsdate(obsdate)
+                    Sunspotter NOAA  HEK NOAA  Sunspotter Longitude  HEK Longitude  Sunspotter Latitude  HEK Latitude
+        id_filename
+        12961                  10838     10838             75.808901             88            16.218928            15
+        12962                  10840     10840             18.191948             26            -2.875433            -3
+        12963                  10841     10841             11.644637             16            11.521983            12
+        12964                  10843     10843             -7.105894             -2            12.301117            12
+        12965                  10844     10844             76.811298             81           -14.756412           -14
+        """
+        obsdate = self.get_nearest_observation(obsdate)
+
+        a = hek.attrs
+        client = hek.HEKClient()
+
+        tstart = datetime.strptime(obsdate, fmt)
+        tend = datetime.strptime(obsdate, fmt) + timedelta(days=days_delta)
+
+        result = client.search(a.Time(tstart, tend), a.AR, a.FRM.Name == noaa_ar)
+
+        data = result['hgs_x', 'hgs_y', 'ar_noaanum']
+        data = data.to_pandas()
+
+        properties = self.get_all_properties_from_obsdate(obsdate)
+
+        if isinstance(properties, pd.Series):
+            properties = pd.DataFrame(data=[properties.values],
+                                      columns=properties.index.values)
+
+        rotated = self.rotate_to_midnight(obsdate)
+
+        lon = [coord[0].value for coord in rotated]
+        lat = [coord[1].value for coord in rotated]
+
+        df_1 = pd.DataFrame(columns=['lon', 'lat'])
+        df_1['lon'] = lon
+        df_1['lat'] = lat
```

Comment:
Maybe we don't need the intermediate `lon`, `lat`
```suggestion
        df_1 = pd.DataFrame(rotated, columns=['lon', 'lat'])
```